### PR TITLE
Add functionality for navigating based on deeplink in notification.

### PR DESCRIPTION
**Pull request by rahulgi**

### CODE REVIEW

File: src/components/shared/Navigation.tsx [hunk 1]
```diff
@@ -653,10 +664,35 @@ class NavigationView extends React.Component<Props> {
...
   }
 
+  /**
+   * Takes an action based on a notification being opened.
+   */
+  handleNotificationOpen = async (
+    notificationOpen: NotificationOpen | undefined
+  ) => {
+    if (notificationOpen) {
+      console.log(notificationOpen);
```
Comment (tinaroh) [inline src/components/shared/Navigation.tsx]: Remove?

File: src/components/shared/Navigation.tsx [hunk 2]
```diff
@@ -653,10 +664,35 @@ class NavigationView extends React.Component<Props> {
...
+    this.handleNotificationOpen(
+      await firebase.notifications().getInitialNotification()
+    );
   }
 
+  /**
+   * Takes an action based on a notification being opened.
+   */
+  handleNotificationOpen = async (
+    notificationOpen: NotificationOpen | undefined
```
Comment (tinaroh) [inline src/components/shared/Navigation.tsx]: What's the different between | undefined vs ?

**Comment by osdiab** [comment]
No difference although it might be that when you say undefined it has to
explicitly be provided as undefined, I forget if that’s the case

File: src/components/shared/Navigation.tsx (hunk 1 above)
Comment (rahulgi) [reply src/components/shared/Navigation.tsx]: oops

File: src/components/shared/Navigation.tsx (hunk 2 above)
Comment (rahulgi) [reply src/components/shared/Navigation.tsx]: (i think omar's comment was intended to be a reply to this).

Yep - the type is the same, but using `?` allows you to call the method without any arguments, while this forces you to always provide the argument, even if it's `undefined`.

**Pull request merged**